File: sdk/julia/src/cloud_gpu.jl
# Cloud GPU: GPU instance management
struct CloudGpuResource
    http::HttpClient
end

# PUT /api/service/{id}/vms/reboot - Reboot virtual server
function reboot(r::CloudGpuResource; kwargs...)
    params = Dict(kwargs)
    http_put(r.http, "/api/service/{id}/vms/reboot"; params = params)
end

# PUT /api/service/{id}/vms/stop - Stop virtual server
function stop(r::CloudGpuResource; kwargs...)
    params = Dict(kwargs)
    http_put(r.http, "/api/service/{id}/vms/stop"; params = params)
end

# PUT /api/service/{id}/vms/start - Start virtual server
function start(r::CloudGpuResource; kwargs...)
    params = Dict(kwargs)
    http_put(r.http, "/api/service/{id}/vms/start"; params = params)
end

# GET /api/service/{id}/vms/firewall - Get firewall rules
function get_firewall_rules(r::CloudGpuResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/service/{id}/vms/firewall"; params = params)
end

# POST /api/service/{id}/vms/firewall - Add firewall rules
function add_firewall_rules(r::CloudGpuResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/vms/firewall"; params = params)
end

# DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule
function remove_firewall_rule(r::CloudGpuResource; kwargs...)
    params = Dict(kwargs)
    http_delete(r.http, "/api/service/{id}/vms/firewall/{position}"; params = params)
end

File: sdk/rust/src/cloud_gpu.rs
//! Cloud GPU: GPU instance management

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct CloudGpuResource {
    http: Arc<HttpClient>,
}

impl CloudGpuResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// PUT /api/service/{id}/vms/reboot - Reboot virtual server
    pub async fn reboot(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.put("/api/service/{id}/vms/reboot", None).await
    }

    /// PUT /api/service/{id}/vms/stop - Stop virtual server
    pub async fn stop(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.put("/api/service/{id}/vms/stop", None).await
    }

    /// PUT /api/service/{id}/vms/start - Start virtual server
    pub async fn start(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.put("/api/service/{id}/vms/start", None).await
    }

    /// GET /api/service/{id}/vms/firewall - Get firewall rules
    pub async fn get_firewall_rules(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/service/{id}/vms/firewall", None).await
    }

    /// POST /api/service/{id}/vms/firewall - Add firewall rules
    pub async fn add_firewall_rules(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/service/{id}/vms/firewall", None).await
    }

    /// DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule
    pub async fn remove_firewall_rule(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.delete("/api/service/{id}/vms/firewall/{position}", None).await
    }

}
